static/js/components/Loader.js

```diff
@@ -6,22 +6,28 @@ export default class Loader extends React.Component {
   props: {
     loaded: boolean,
     children?: React$Element<*>[],
+    shouldRenderAll?: boolean
   };
 
-  render() {
-    const { loaded, children } = this.props;
+  renderAll = (loaded: boolean, children?: React$Element<*>[]) => ([
+    <div className="loader" style={{'display': !loaded ? "block" : "none"}} key="loader">
+      <Spinner singleColor />
+    </div>,
+    <div style={{'display': loaded ? "block" : "none"}} key="content">
+      {children}
+    </div>
+  ]);
 
-    let content;
-    if (loaded) {
-      content = children;
-    } else {
-      content = <div className="loader">
-        <Spinner singleColor />
-      </div>;
-    }
+  renderLoadedOrSpinner = (loaded: boolean, children?: React$Element<*>[]) => (
+    loaded ? children : <div className="loader"> <Spinner singleColor /> </div>
+  );
 
-    return <div>
-      {content}
-    </div>;
+  render() {
+    const { loaded, children, shouldRenderAll = false } = this.props;
+    return (
+      <div>
+        { shouldRenderAll ? this.renderAll(loaded, children) : this.renderLoadedOrSpinner(loaded, children) }
```

Comment:
I'm guessing any tests where we assert that the children aren't on the DOM will fail, because they're still being rendered (albeit invisibly)